src/workerd/io/io-context.c++

```diff
@@ -543,6 +537,12 @@ kj::Promise<void> IoContext::IncomingRequest::drain() {
     timeoutPromise = timeoutPromise.exclusiveJoin(kj::mv(drainPaf.promise));
   } else {
     // For non-actor requests, apply the configured soft timeout, typically 30 seconds.
+    auto timeoutLogPromise = [this]() -> kj::Promise<void> {
```

Comment:
Nit: a comment here indicating that capturing the `this` is safe would be helpful as a signal to anyone coming into this code later. Non-blocking tho.